libs/angular/src/vault/components/icon.component.ts

```diff
@@ -44,8 +47,16 @@ export class IconComponent implements OnChanges {
     // to avoid this we reset all state variables.
     this.image = null;
     this.fallbackImage = null;
-    this.imageEnabled = !(await this.stateService.getDisableFavicon());
-    this.load();
+
+    this.settingsService.disableFavicon$.pipe(first(), takeUntil(this.destroy$)).subscribe((v) => {
+      this.imageEnabled = !v;
+      this.load();
+    });
```

Comment:
We should be able to simplify this and use `firstValueFrom` here. Which should also let us get rid of the destroy subject.

```suggestion
    this.imageEnabled = !await firstValueFrom(this.settingsService.disableFavicon);
    this.load();
```

Reply:
This turns it into a promise, which because of the OnPush Change detection, it never reacts to the changes. It needs to get the event from the observable in order to change.

Reply:
Not following, maybe I'm missing something. `ngOnChanges` is executed when the input is changed. Both `o$.pipe(first())` and `firstValueFrom(o$)` retrieves only the first value from an observable. Since the observable is a subject it will always have a value and will immediately resole with the setting.

If you want it to continue to react on changes in the `disabledFavicon$` observable, instead of just getting the first value, the observable needs to be moved into the `ngOnInit` and `first()` needs to be removed so that it is notified about further messages.

Reply:
Moved it to ngOnInit, thank you!